### DISCUSSION TITLE
Integrate Image viewer

### DISCUSSION THREAD

**Discussion by slowflyer74** [kedder/openvario-shell#50] [Ideas]
As an Aerobatic Pilot, I would like an image viewer application that displays an image stored on an USB stick or locally that contains the aerobatic program (instead of gluing a piece of paper over the OV display).
Optional: paging with remotestick keys or rotary encoder device(left/right)
Optional: Zooming with remotestick keys or rotary encoder device (up/down)
During aerobatic, XCSoar is not needed.

What do you think about this?

**Comment by kedder** [comment]
@slowflyer74 This sounds reasonable. It should be possible to implement as a separate app, in a similar way [openvario-compman](https://github.com/kedder/openvario-compman) is implemented.

Are you willing to contribute the code?

**Reply by slowflyer74** [reply]
I can try. Will have a look at compman integration and do my best.

**Reply by kedder** [reply]
Cool, you can ask me anything. Putting an item should be pretty trivial. The trickiest part will be figuring out how to actually display the image on the screen.

**Reply by slowflyer74** [reply]
I tried several tools to output pixels into the console, but this did not result in a readable image... . Maybe PyGame could be useful, but I was not yet able to compile it on the cubieboard. Maybe you have more experience doing this?

**Reply by kedder** [reply]
I know  that it is possible, because kernel manages to output the splash screen. I'm not completely sure what infrastructure is involved in it, but I suspect you can output pixels directly to the linux framebuffer (`/dev/fb0` or smth). I can try to find out how to do that exactly in the evening.

This is pretty low-level approach though and will require decoding various graphics formats (jpg, png, etc) into pixels and resizing the image to fit on the screen. The alternative is to run some graphics server, like wayland and run normal image viewer on it, but this is also not that simple - requires to include a quite large dependency into the image. I have a feeling that the former will be easiler to implement.

**Reply by slowflyer74** [reply]
somting like psplash? Unfortunately the image is compiled into the binary, so it's not possible to exchange it easily. All other tools like pygame, mplayer, fbi a.s.o all need FreeType2 installed, but this seems not to be available for the OV.

**Reply by kedder** [reply]
Quick googling also revealed tools like [fim](http://manpages.ubuntu.com/manpages/bionic/man1/fim.1.html) and [fbvs](https://github.com/kaihs/fbvs). I wonder if it is possible to compile them for openvario. Having freetype2 installed is not a huge deal (except I'm not sure how much other dependencies it pulls). I can try compiling them and putting to my openvario PPA.

**Comment by kedder** [comment]
Alright, I was able to build `fbida` package, that includes `fbi` image viewer. And it seems to work! I was able to show .bmp images on the screen (didn't try other formats).

You can install it from my PPA (you already have it configured if you have installed ovshell:

```
opkg update
opkg install fbida
```

Then you should have `fbi` binary. Use it like this:

```
fbi *.bmp
```

I believe it should support png and jpg images as well.

**Reply by slowflyer74** [reply]
It works, also with gifs. Thanks a lot
